src/Microsoft.Azure.SignalR.Management/DependencyInjectionExtensions.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System.ComponentModel;
+using System.Linq;
+using System.Reflection;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.SignalR.Management
+{
+    internal static class DependencyInjectionExtensions //TODO: not ready for public use
+    {
+        /// <summary>
+        /// Add required dependencies according to the configured options <see cref="ServiceManagerOptions"/>. Please call this method after <see cref="ServiceManagerOptions"/> is configured.
+        /// </summary>
+        public static IServiceCollection AddSignalRServiceManager(this IServiceCollection services)
+        {
+            services.PostConfigure<ServiceManagerOptions>(o => o.ValidateOptions());
+            services.AddSingleton<ServiceManagerContextSetup>()
+                .AddSingleton<IConfigureOptions<ServiceManagerContext>>(sp => sp.GetService<ServiceManagerContextSetup>());
+            services.AddSingleton<ServiceOptionsSetup>()
+                .AddSingleton<IConfigureOptions<ServiceOptions>>(sp => sp.GetService<ServiceOptionsSetup>());
+            if (services.Any(descriptor => descriptor.ServiceType == typeof(IOptionsChangeTokenSource<ServiceManagerOptions>)))
+            {
+                services.AddSingleton<IOptionsChangeTokenSource<ServiceManagerContext>>(sp => sp.GetService<ServiceManagerContextSetup>());
+                services.AddSingleton<IOptionsChangeTokenSource<ServiceOptions>>(sp => sp.GetService<ServiceOptionsSetup>());
+            }
+            services.WithCallingAssembly(overwrite: false);
```

Comment:
why call it by default? only Function extension wants **calling** assembly I think, others want **current** Management assembly